sdk/servicebus/azure-servicebus/azure/servicebus/_common/_connection_string_parser.py

```diff
@@ -0,0 +1,67 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from azure.servicebus.management._models import DictMixin
+
+class ServiceBusConnectionStringProperties(DictMixin):
+    """
+    Properties of a connection string.
+    """
+    def __init__(self, **kwargs):
+        self.fully_qualified_namespace = kwargs.pop('fully_qualified_namespace', None)
+        self.endpoint = kwargs.pop('endpoint', None)
+        self.entity_path = kwargs.pop('entity_path', None)
+        self.shared_access_signature = kwargs.pop('shared_access_signature', None)
+        self.shared_access_key_name = kwargs.pop('shared_access_key_name', None)
+        self.shared_access_key = kwargs.pop('shared_access_key', None)
```

Comment:
are they read-only? if they're read only, then we should consider making them being properties.

if not, then we'd better mention them as `ivar` in the doc string.

Reply:
It's unfortunate boilerplate but I do like the thought of starting read-only, lets us add writeable later and keeps this constrained, as was our initial goal/concern about the full builder proposals.

Reply:
they should be readonly - changing to properties